dotcom-rendering/src/model/enhanceTimeline.test.ts

```diff
@@ -0,0 +1,84 @@
+import assert from 'node:assert/strict';
```

Comment:
While I think it’s really nifty to use the [native Node test runner](https://nodejs.org/docs/latest-v20.x/api/test.html), I think it’s best to do this in a separate PR and keep Jest for consistency?

Reply:
These aren't used for testing our application code, they're included primarily as a form of type guard, for example to prevent TS errors when indexing into arrays. I'm not sure using something like Jest's `expect` will do this (i.e. narrow the types)?

I think `assert` should be preferable to using a type cast, as the error will hopefully be more accurate; type casts often result in unexpected behaviour, or errors downstream that are harder to trace back to their origin.

If there's another way to do this within Jest I'm happy to switch to that!

Reply:
That’s a perfectly valid use-case and [definitely superior to casting](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e66570914d31929401f536a7d70a46ef8e290b56/types/node/assert.d.ts#L12)… I wrongly assumed this was related to testing, but it’s [a distinct module](https://nodejs.org/docs/latest-v20.x/api/assert.html).

The [`console.assert()`](https://developer.mozilla.org/en-US/docs/Web/API/console/assert_static) method will not throw an error in the same way and cannot be used instead.